src/NHibernate/Linq/DmlExpressionRewriter.cs

```diff
@@ -38,6 +38,30 @@ void AddSettersFromBindings(IEnumerable<MemberBinding> bindings, string path)
 				}
 		}
 
+		void AddSettersFromConstructorArguments(NewExpression newExpression, string path)
+		{
+			var argumentDefinitions = newExpression.Constructor.GetParameters();
```

Comment:
Thanks for the pointer @hazzik ,
So instead it should be `newExpression.Members` here. At least it is documented as matching constructor arguments, but that logic still looks quite brittle.